### DISCUSSION TITLE
ZCU102 deployment error

### DISCUSSION THREAD

**Discussion by diracpulse** [Xilinx/finn#584] [Q&A]
Hello everyone, 

I have a ZCU102 on which I would like to deploy my FINN model. As you know, the ZCU102 is not supported out of the box, so I have built a PYNQ image for it. The discussion on how it was built is [here](https://discuss.pynq.io/t/pynq-error-with-trace-cntrl-32/3923). Please don't be too intolerant on the fact that I had forgotten to patch the 2yk22 bug on that specific Vivado version ... I know it was kinda of an epic-fail. 

As far as I can tell, the PYNQ image seems fully functional, I am able to ssh to the board and I have generated the ssh-keys, as specified [in the getting-started tutorial](https://finn.readthedocs.io/en/latest/getting_started.html#vivado-vitis-license) . I ran into the same problem reported in discussion #442 and have applied the same solution successfully. 

Some steps later, in the tfc_end2end_example notebook, I run into the error reported in the image below : 

![image](https://user-images.githubusercontent.com/5065137/161266198-653fc7d2-544c-4078-b578-aa51eb4a1969.png)

This is the metadata of the model : 

![image](https://user-images.githubusercontent.com/5065137/161265790-1d4eb179-7571-4ed8-bd28-466a79ea102e.png)

It is worthy of notice, that I replaced the user@ip parameters with their manual counterpart. I suppose the DeployToPynq function is not working properly, since I see no folder being created on the target. The error reported above cascades a couple of cells after, as the error reported in the screenshot below : 

![image](https://user-images.githubusercontent.com/5065137/161266090-f05ec9a2-b113-46ca-bbb2-5219c94c0b17.png)

What is really strange imho, is that the folders with the "model-reference" do not have the same value.

I would be really glad to get some suggestions and sorry again for the beginner question

**Comment by fpjentzsch** [comment]
Hi, this seems similar to the (unsolved) error discussed in the later messages of this thread: https://github.com/Xilinx/finn/discussions/389

With "model-reference", do you mean that "c6v9ps2b" ist not "zdf03n3l" in your case? Maybe you re-ran some earlier cells, triggering generation of a new randomly-named deployment folder?

Could you perform the copy operation within DeployToPYNQ ([here](https://github.com/Xilinx/finn/blob/26169798b1231520b8632a042c587833a0afdc0b/src/finn/transformation/fpgadataflow/make_deployment.py#L110)) manually to figure out if this is truly a bug or just a permissions issue of some sort?
